### PR TITLE
Add external networking edges import

### PR DESCRIPTION
**Pull request by robinschmid**
- Import external edges
- show full network with external edges and feature shape corrlation edges

### CODE REVIEW

**Review by SteffenHeu — COMMENTED** [review]
I've had a first look, but this is a good addition. Maybe we should add this to project load/save to store the IIN networks.

File: src/main/java/io/github/mzmine/modules/dataprocessing/group_metacorrelate/export/ExportCorrAnnotationTask.java [hunk 1]
```diff
@@ -65,517 +63,541 @@
...
-                                    Integer id2 = renumbered.get(getRowMapKey(e.getKey()));
-
-                                    // the data
-                                    exportEdge(ann, "MS1 annotation", id1, id2, corrForm.format((id.getScore() + adduct.getScore()) / 2d), //
-                                            id.getAdduct() + " " + adduct.getAdduct() + " dm/z=" + mzForm.format(dmz));
-                                    added.incrementAndGet();
-                                }
-                            });
-                        });
+    NumberFormat mzForm = MZmineCore.getConfiguration().getMZFormat();
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/group_metacorrelate/export/ExportCorrAnnotationTask.java:526]: changed them all. Good catch

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableContextMenu.java [hunk 2]
```diff
@@ -332,9 +333,14 @@ private void initSearchMenu() {
 
   private void initShowMenu() {
 
-    final MenuItem showNetworkVisualizerItem = new ConditionalMenuItem("Feature overview (network)",
-        () -> hasMs2(selectedRows));
-    showNetworkVisualizerItem.setOnAction(e -> showNetworkVisualizer());
+    final MenuItem showNetworkVisualizerItem = new ConditionalMenuItem(
+        "Feature overview (IIMN network)", () -> hasMs2(selectedRows));
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableContextMenu.java:337]: True.

[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableContextMenu.java:337]: Its okay that I added the items for full network and IIMN network (nearest neighbors)

**Comment by robinschmid** [comment]
Not sure if this fully captures the relationship... but maybe can be expanded for this

**Comment by robinschmid** [comment]
@roman-bushuiev did you try this?

**Comment by robinschmid** [comment]
This should be merged soon

**Comment by roman-bushuiev** [comment]
I will try it this week. Is it also possible to import custom annotations for nodes (e.g. SMILES or some numerical values)?

**Comment by robinschmid** [comment]
Did not see your message.
So far it is not possible. 
The best option would be to make a small module to import results from your tool either from csv files, a folder, or json files.
This is super easy. I can guide you.

e.g spectral library matches are covered in this class:
SpectralLibraryMatchesType

Once this is done it is automatically reflected in the table and in exports etc.

File: src/main/java/io/github/mzmine/datamodel/features/FeatureList.java [hunk 3]
```diff
@@ -407,28 +392,36 @@ default R2RMap<RowsRelationship> getMs1CorrelationMap() {
...
+    return getRowMap(relationship.toString());
+  }
+
+  /**
+   * A mutable map of row-to-row relationships.
+   *
+   * @param type the relationship between two rows
+   * @return
+   */
+  default Optional<R2RMap<RowsRelationship>> getRowMap(String type) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/FeatureList.java:415]: Why string, why not an enum for the type?

**Pull request merged**